<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RemoteTCPSinkSettingsDialog</class>
 <widget class="QDialog" name="RemoteTCPSinkSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>409</width>
    <height>947</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="serverSettingsGroup">
     <property name="title">
      <string>Server Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="1">
       <widget class="QSpinBox" name="maxSampleRate">
        <property name="toolTip">
         <string>Maximum channel sample rate</string>
        </property>
        <property name="minimum">
         <number>2700</number>
        </property>
        <property name="maximum">
         <number>100000000</number>
        </property>
        <property name="value">
         <number>10000000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="timeLimitLabel">
        <property name="text">
         <string>Time Limit</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="maxClientsLabel">
        <property name="text">
         <string>Max Clients</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="timeLimit">
        <property name="toolTip">
         <string>Connection time limit in minutes if max clients reached. 0 for no limit.</string>
        </property>
        <property name="maximum">
         <number>100000</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="maxSampleRateLabel">
        <property name="text">
         <string>Max Ch. Sample Rate</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="maxSampleRateUnits">
        <property name="text">
         <string>S/s</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="maxClients">
        <property name="toolTip">
         <string>Maximum number of simultaneous clients</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="timeLimitUnits">
        <property name="text">
         <string>mins</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="iqOnlyLabel">
        <property name="text">
         <string>IQ only</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="iqOnly">
        <property name="toolTip">
         <string>Transmit uncompressed IQ only. Disables compression, position and messaging support.</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="compressionSettings">
     <property name="title">
      <string>Compression</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="compressorLabel">
        <property name="text">
         <string>Compressor</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="compressor">
        <item>
         <property name="text">
          <string>FLAC</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>zlib</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="compressionLevelLabel">
        <property name="text">
         <string>Compression Level</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="compressionLevel">
        <property name="toolTip">
         <string>0 - Least compression. 8 - Most compression Higher compression requires more CPU.</string>
        </property>
        <property name="maximum">
         <number>8</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="blockSizeLabel">
        <property name="text">
         <string>Block size</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="blockSize">
        <item>
         <property name="text">
          <string>4096</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>16384</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="sslSettingsGroup">
     <property name="title">
      <string>SSL Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QLineEdit" name="certificate">
        <property name="toolTip">
         <string>SSL certificate</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QToolButton" name="browseKey">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/load.png</normaloff>:/load.png</iconset>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="keyLabel">
        <property name="text">
         <string>Key</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="key">
        <property name="toolTip">
         <string>SSL certificate key</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="certificateLabel">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Certificate</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QToolButton" name="browseCertificate">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/load.png</normaloff>:/load.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="directorySettings">
     <property name="title">
      <string>Public Directory</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="3" column="0">
       <widget class="QLabel" name="antennaLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Antenna</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="minFrequency">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Minimum recommend frequency in MHz</string>
        </property>
        <property name="maximum">
         <number>20000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSpinBox" name="publicPort">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Publically accessible port number</string>
        </property>
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>1234</number>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="directionLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Direction</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="3">
       <widget class="QLineEdit" name="location">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Town and country where antenna is located</string>
        </property>
        <property name="maxLength">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="publicListingLabel">
        <property name="text">
         <string>List Server </string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="isotropicLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Isotropic</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1" colspan="3">
       <layout class="QHBoxLayout" name="directionLayout">
        <item>
         <widget class="QLabel" name="azimuthLabel">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Az</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="azimuth">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="statusTip">
           <string>Antenna azimuth in degrees</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>360.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="elevationLabel">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>El</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="elevation">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="statusTip">
           <string>Antenna elevation in degrees</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-90.000000000000000</double>
          </property>
          <property name="maximum">
           <double>90.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="2" column="2">
       <widget class="QSpinBox" name="maxFrequency">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Maximum recommend frequency in MHz</string>
        </property>
        <property name="maximum">
         <number>20000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="publicAddressLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Address</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="locationLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Location</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLabel" name="frequencyUnits">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>MHz</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="publicAddress">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Publically accessible IP address or hostname</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="isotropic">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Check to indicate an antenna that is isotropic (non-directional)</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="publicListing">
        <property name="toolTip">
         <string>Whether to list the server as publically accessible</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="frequencyLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Frequency Range</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="3">
       <widget class="QLineEdit" name="antenna">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Antenna description</string>
        </property>
        <property name="maxLength">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="rotatorLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Rotator</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1" colspan="2">
       <widget class="QComboBox" name="rotator">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Rotator feature to get antenna direction from</string>
        </property>
        <item>
         <property name="text">
          <string>None</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="ipBlacklistGroup">
     <property name="title">
      <string>IP Blacklist</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QListWidget" name="ipBlacklist">
        <property name="toolTip">
         <string>List of IP addresses from which connections should not be allowed</string>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::MultiSelection</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="ipBlacklistLayout">
        <item>
         <widget class="QToolButton" name="addIP">
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="removeIP">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>maxClients</tabstop>
  <tabstop>timeLimit</tabstop>
  <tabstop>maxSampleRate</tabstop>
  <tabstop>iqOnly</tabstop>
  <tabstop>compressor</tabstop>
  <tabstop>compressionLevel</tabstop>
  <tabstop>blockSize</tabstop>
  <tabstop>certificate</tabstop>
  <tabstop>browseCertificate</tabstop>
  <tabstop>key</tabstop>
  <tabstop>browseKey</tabstop>
  <tabstop>publicListing</tabstop>
  <tabstop>publicAddress</tabstop>
  <tabstop>publicPort</tabstop>
  <tabstop>minFrequency</tabstop>
  <tabstop>maxFrequency</tabstop>
  <tabstop>antenna</tabstop>
  <tabstop>location</tabstop>
  <tabstop>isotropic</tabstop>
  <tabstop>rotator</tabstop>
  <tabstop>azimuth</tabstop>
  <tabstop>elevation</tabstop>
  <tabstop>ipBlacklist</tabstop>
  <tabstop>addIP</tabstop>
  <tabstop>removeIP</tabstop>
 </tabstops>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RemoteTCPSinkSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RemoteTCPSinkSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
